src/pages/Auth/Auth.js

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useLocation } from "react-router-dom";
+import { REDIRECT_URI } from "../../config";
+
+const Auth = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const userAuthCode = location.search.split("code=")[1];
+  const kakaoAuthUrl = `https://kauth.kakao.com/oauth/token?grant_type=authorization_code&client_id=${process.env.REACT_APP_CLIENT_ID}&redirect_uri=${REDIRECT_URI}&code=${userAuthCode}`;
+
+  useEffect(() => {
+    fetch(kakaoAuthUrl, {
+      method: "POST",
+      headers: {
+        "Content-type": "application/x-www-form-urlencoded;charset=utf-8",
+      },
+    })
+      .then(res => {
+        if (res.status === 200) {
+          return res.json();
+        } else {
+        }
+      })
+      .then(data => localStorage.setItem("access_token", data.access_token))
+      .then(() => navigate("/"));
```

Comment:
```suggestion
      .then(data => {
      localStorage.setItem("access_token", data.access_token)
      navigate('/')
      })
```
* 첫번째 응답에서 통신이 성공한 json을 객체로 변환했더라도, 
* 통신은 항상 성공한다는 보장을 하기어렵고, 중간에 인터넷이 끊기거나, 속도가 갑자기 느려져서 언제 응답을 받을 수 있을 지 같은 변수가 작용할수도 있어서 언제 종료 될 지 예측하기 어렵습니다. 
* 그래서 반환된 status code로 분기처리 하는 로직으로 수정 해 주세요!
* navigate는 토큰 저장한 다음 호출해버리면 됩니다!

Reply:
넵! 수정하였습니다. 감사합니다!